keras/utils/generic_utils.py

```diff
@@ -313,8 +313,13 @@ def update(self, current, values=None, force=False):
                         bar += '='
                 bar += ('.' * (self.width - prog_width))
                 bar += ']'
-                sys.stdout.write(bar)
-                self.total_width = len(bar)
+            else:
+                numdigits = 7
+                barstr = '%%%dd/Unknown' % numdigits
+                bar = barstr % current
```

Comment:
Don't format a string two lines, do it in one pass. Also, there is seemingly no need for the `% 7`...

Reply:
Fixed. Just wanted to keep both branches of the conditional as identical as possible.